### PR TITLE
[5.0] add BaseRequestHandler to MonitoringHandler

### PR DESCRIPTION
**Pull request by TaykYoku**
BEGINRELEASENOTES

NEW: add BaseRequestHandler to MonitoringHandler

ENDRELEASENOTES

### CODE REVIEW

File: src/WebAppDIRAC/WebApp/handler/MonitoringHandler.py [hunk 1]
```diff
@@ -55,21 +55,19 @@ def web_getSelectionData(self, type):
             length = len(retVal["Value"][record])
             if length > 10000:
                 records[record] = retVal["Value"][record][length - 5000 :]
-                message = (
-                    "The %s accounting type contains to many rows: %s - > %d. Note: Only 1000 rows are returned!"
-                    % (type, record, length)
-                )
+                message = f"The {monitoringType} monitoring type contains too many rows: {record} -> {length}"
+                message += " Note: Only 1000 rows are returned!"
```
Comment (andresailer) [inline src/WebAppDIRAC/WebApp/handler/MonitoringHandler.py]:
```suggestion
                message += " Note: Only 5000 rows are returned!"
```
That's what the slicing in line 57 does, right?

File: src/WebAppDIRAC/WebApp/handler/MonitoringHandler.py [hunk 2]
```diff
@@ -179,62 +165,39 @@ def __parseFormParams(self):
...
-        """
-        callback = {}
-        if "file" not in self.request.arguments:
-            callback = {"success": "false", "error": "Maybe you forgot the file?"}
-            self.finish(callback)
-            return
-        plotImageFile = self.get_argument("file")
+    def web_getPlotImg(self, fileName=None):
+        """Get plot image"""
+        if fileName:
```
Comment (andresailer) [inline src/WebAppDIRAC/WebApp/handler/MonitoringHandler.py]:
```suggestion
        if not fileName:
```

File: src/WebAppDIRAC/WebApp/handler/MonitoringHandler.py [hunk 3]
```diff
@@ -179,62 +165,39 @@ def __parseFormParams(self):
...
-        """
-        callback = {}
-        if "file" not in self.request.arguments:
-            callback = {"success": "false", "error": "Maybe you forgot the file?"}
-            self.finish(callback)
-            return
-        plotImageFile = self.get_argument("file")
+    def web_getPlotImgFromCache(self, fileName=None):
+        """Get plot image from cache."""
+        if plotImageFile := fileName:
```
Comment (andresailer) [inline src/WebAppDIRAC/WebApp/handler/MonitoringHandler.py]: Some `not` missing?

File: src/WebAppDIRAC/WebApp/handler/MonitoringHandler.py [hunk 4]
```diff
@@ -179,62 +165,39 @@ def __parseFormParams(self):
...
+            return {"success": "false", "error": "Maybe you forgot the file?"}
 
         retVal = extractRequestFromFileId(plotImageFile)
         if not retVal["OK"]:
-            callback = {"success": "false", "error": retVal["Value"]}
-            self.finish(callback)
-            return
+            return {"success": "false", "error": retVal["Value"]}
```
Comment (andresailer) [inline src/WebAppDIRAC/WebApp/handler/MonitoringHandler.py]:
```suggestion
            return {"success": "false", "error": retVal["Message"]}
```
Unless that function does some special return.

Checked: this does return S_ERROR https://github.com/DIRACGrid/DIRAC/blob/integration/src/DIRAC/Core/Utilities/Plotting/FileCoding.py#L15

File: src/WebAppDIRAC/WebApp/handler/MonitoringHandler.py [hunk 5]
```diff
@@ -247,35 +210,23 @@ def web_getPlotImgFromCache(self):
 
         retVal = codeRequestInFileId(fields)
         if not retVal["OK"]:
-            callback = {"success": "false", "error": retVal["Value"]}
-            self.finish(callback)
-            return
+            return {"success": "false", "error": retVal["Value"]}
```
Comment (andresailer) [inline src/WebAppDIRAC/WebApp/handler/MonitoringHandler.py]:
```suggestion
            return {"success": "false", "error": retVal["Message"]}
```
But https://github.com/DIRACGrid/DIRAC/blob/c165c81c85f2352aed23baeda1160d6a3f6c5f3c/src/DIRAC/Core/Utilities/Plotting/FileCoding.py#L15 never returns S_ERROR

File: src/WebAppDIRAC/WebApp/handler/MonitoringHandler.py [hunk 6]
```diff
@@ -179,62 +165,39 @@ def __parseFormParams(self):
...
     def web_generatePlot(self):
-        callback = {}
-        retVal = yield self.threadTask(self.__queryForPlot)
-        if retVal["OK"]:
-            callback = {"success": True, "data": retVal["Value"]["plot"]}
-        else:
-            callback = {"success": False, "errors": retVal["Message"]}
-        self.finish(callback)
+        if (retVal := self.__queryForPlot())["OK"]:
+            return {"success": True, "data": retVal["Value"]["plot"]}
```
Comment (andresailer) [inline src/WebAppDIRAC/WebApp/handler/MonitoringHandler.py]: The other returns always use string "true", "false", is that equivalent with the booleans?

**Review by andresailer — CHANGES_REQUESTED** [review]
Few things need changes

**Comment by fstagni** [comment]
@andresailer I think I addressed all.

**Review by andresailer — APPROVED** [review]

**Pull request merged**